src/Hooks.ts

```diff
@@ -17,18 +17,69 @@ import {
   EventType,
 } from 'react-navigation';
 
-export function useNavigation<S>(): NavigationScreenProp<S & NavigationRoute> {
+function useNavigationSafe<S>(): NavigationScreenProp<S & NavigationRoute> {
   const navigation = useContext(NavigationContext) as any; // TODO typing?
+
   if (!navigation) {
     throw new Error(
       "react-navigation hooks require a navigation context but it couldn't be found. " +
         "Make sure you didn't forget to create and render the react-navigation app container. " +
         'If you need to access an optional navigation object, you can useContext(NavigationContext), which may return'
     );
   }
+
   return navigation;
 }
 
+function useStableCallback(cb: (...args: any) => any) {
+  const ref = useRef(cb);
+
+  useLayoutEffect(() => {
+    ref.current = cb;
+  }, [cb]);
+
+  return useCallback((...args) => ref.current(...args), [ref]);
+};
+
+// This is intended to solve: https://github.com/react-navigation/hooks/issues/40
+function useStableActions<S>(navigation: NavigationScreenProp<S & NavigationRoute>) {
+  return {
+    // common actions
+    navigate: useStableCallback(navigation.navigate),
+    goBack: useStableCallback(navigation.goBack),
+    addListener: useStableCallback(navigation.addListener),
+    isFocused: useStableCallback(navigation.isFocused),
+    setParams: useStableCallback(navigation.setParams),
+    getParam: useStableCallback(navigation.getParam),
+    dispatch: useStableCallback(navigation.dispatch),
+    dangerouslyGetParent: useStableCallback(navigation.dangerouslyGetParent),
+    isFirstRouteInParent: useStableCallback(navigation.isFirstRouteInParent),
+
+    // drawer navigator, actions
+    openDrawer: useStableCallback(navigation.openDrawer),
```

Comment:
before, on a stack navigation, openDrawer would be undefined. Now it's a callback that will crash with "undefined is not a function", not exactly the same ;)

Reply:
You are right! On v4 this also change since StackNavigator is in another package. I'll take a look on this.